### PR TITLE
build: update llvm tag to 4546397e

### PR DESCRIPTION
**Pull request by ramiro050**
This commit makes the following changes needed to update bump LLVM:

- Replace `linalg.init_tensor` with `tensor.empty` (see: https://reviews.llvm.org/D135129)
- Replace `NoSideEffect` with `Pure` (see https://reviews.llvm.org/D135505)
- Replace `body` region accessor for `ReduceOp` and `ReduceWindowOp` with `getBody`
- Fix incorrect use of `tosa::ReduceSumOp` in `AtenNativeLayerNormOp` conversion pattern. The result type of `tosa::ReduceSumOp` must have the same rank as the input type. (see:
https://www.mlplatform.org/tosa/tosa_spec.html#_reduce_sum)

Co-authored-by: Ashay Rane <ashay@users.noreply.github.com>

### CODE REVIEW

**Review by sjain-stanford — APPROVED** [review]
LGTM!

File: lib/Conversion/TorchToLinalg/TensorScalarInterop.cpp [hunk 1]
```diff
@@ -178,7 +178,8 @@ class ConvertPrimNumToTensorScalarOp
     Location loc = op.getLoc();
     Value a = adaptor.a();
     Value outTensor =
-        rewriter.create<linalg::InitTensorOp>(loc, ValueRange{}, a.getType())
+        rewriter
+            .create<tensor::EmptyOp>(loc, ArrayRef<OpFoldResult>{}, a.getType())
```
[the remark this answers is not in the document]
Comment (ramiro050) [reply lib/Conversion/TorchToLinalg/TensorScalarInterop.cpp:182]: I think so, I just ran `git clang-format HEAD~1` and no files were modifed

File: e2e_testing/xfail_sets.py [hunk 2]
```diff
@@ -457,6 +457,13 @@
     "_LogSoftmaxModuleStable_basic",
     "LiftFreshCopyModule_basic",
     "ReduceSumDimIntListKeepDimNegativeDimStaticModule_basic",
+    "ReduceSumDimIntListFloatModule_basic",
+    "ReduceSumDimIntListIntModule_basic",
+    "ReduceSumDimIntListKeepDimFloatModule_basic",
+    "ReduceSumDimIntListKeepDimIntModule_basic",
+    "ReduceSumFloatModule_basic",
+    "ReduceSumSignedIntModule_basic",
+    "ReduceSumUnsignedIntModule_basic",
```
[the remark this answers is not in the document]
Comment (ramiro050) [reply e2e_testing/xfail_sets.py:466]: Oh no, these are new tests that pass now. For TOSA, the set in this file represents the tests expected to pass

**Comment by ramiro050** [comment]
> LGTM! Thanks for fixing the associated issues upstream as well!

Thank you for the help!

**Pull request merged**